vissl/engines/benchmark_suite_scheduler.py

```diff
@@ -0,0 +1,545 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import collections
+import json
+import os
+import time
+from pathlib import Path
+from typing import List
+
+import submitit
+from fvcore.common.file_io import PathManager
+from hydra.experimental import compose, initialize_config_module
+from vissl.config.attr_dict import AttrDict
+from vissl.trainer.trainer_main import build_task
+from vissl.utils.distributed_launcher import launch_distributed_on_slurm
+from vissl.utils.hydra_config import convert_to_attrdict, is_hydra_available
+from vissl.utils.io import load_file
+from vissl.utils.misc import flatten_dict, retry
+from vissl.utils.slurm import is_submitit_available
+
+
+# How many times to retry a slurm job submission.
+_NUM_SLURM_RETRIES = 5
+# How many seconds to sleep between iterations of the main loop.
+_SLEEP_TIME_SECONDS = 15
+# Slurm states marked as terminal. SlurmEvulator#evaluate will finish
+# once all jobs are in a terminal state.
+_SLURM_JOB_TERMINAL_STATES = [
+    "BOOT_FAIL",
+    "CANCELLED",
+    "COMPLETED",
+    "DEADLINE",
+    "FAILED",
+    "NODE_FAIL",
+    "OUT_OF_MEMORY",
+    "REVOKED",
+    "SPECIAL_EXIT",
+    "STOPPED",
+    "SUSPENDED",
+    "TIMEOUT",
+]
+# Wait for the training checkpoint folder to be available for 1 hour.
+_TRAINING_CHECKPOINT_FILE_WAIT_SECONDS = 60 * 60
```

Comment:
this is a bit tricky. I think we shouldn't enforce this strictly and allow to override. for instance, for our SEER model, sometimes checkpoints appear every 2 hours etc

Reply:
Ah this is confusingly named. This is the amount of time we are waiting for the training folder to be present, in order to read it's YAML config.